flink-core/src/test/java/org/apache/flink/api/connector/source/mocks/MockSplitEnumeratorContext.java

```diff
@@ -120,7 +123,14 @@ public void assignSplits(SplitsAssignment<SplitT> newSplitAssignments) {
     }
 
     @Override
-    public void signalNoMoreSplits(int subtask) {}
+    public void signalNoMoreSplits(int subtask) {
+        subtaskHasNoMoreSplits[subtask] = true;
+    }
+
+    @Override
+    public void signalIntermediateNoMoreSplits(int subtask) {
+        subtaskHasNoMoreSplits[subtask] = false;
```

Comment:
This method should be empty. The values should be false initially but not set to be false in this method.